### PR TITLE
Download openblas binary from GH repo

### LINKED ISSUE #15854: CI: Windows Meson job failing sometimes on OpenBLAS binary download
I've seen this a bunch of times now:
```
 Installing collected packages: certifi, urllib3, tqdm, soupsieve, six, PySocks, idna, filelock, charset-normalizer, requests, beautifulsoup4, gdown
Successfully installed PySocks-1.7.1 beautifulsoup4-4.10.0 certifi-2021.10.8 charset-normalizer-2.0.12 filelock-3.6.0 gdown-4.4.0 idna-3.3 requests-2.27.1 six-1.16.0 soupsieve-2.3.1 tqdm-4.63.0 urllib3-1.26.9
C:\hostedtoolcache\windows\Python\3.9.10\x64\lib\site-packages\gdown\cli.py:127: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  warnings.warn(

 	Cannot retrieve the public link of the file. You may need to change
	the permission to 'Anyone with the link', or have had many accesses. 

You may still be able to access the file from the browser:

	 https://drive.google.com/uc?id=1G3jwxMY4ZVRqvz6owl0kmw_Innmr4ZnV 

Access denied with the following error:
unzip:  cannot find either openblas_32_if.zip or openblas_32_if.zip.zip.
```

~20% of the actions on the first page of https://github.com/scipy/scipy/actions/workflows/windows.yml show this failure. @matthew-brett it'd be nice if we could move this, or could make the download more robust somehow. What do you think?

### PR DESCRIPTION
**Pull request by jarrodmillman**
Fix #15854

### CODE REVIEW

**Comment by jarrodmillman** [comment]
Testing download from my branch. Before merging, we should update it to use the scipy/scipy-ci main branch.

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -48,9 +48,9 @@ jobs:
           # Download and install pre-built OpenBLAS library
           # Built with mingw-w64, -ucrt -static.
           # https://github.com/matthew-brett/openblas-libs/blob/ucrt-build/build_openblas.ps1
-          choco install unzip
-          pip install gdown
-          gdown --id 1G3jwxMY4ZVRqvz6owl0kmw_Innmr4ZnV
+          choco install unzip -y
+          choco install wget -y
+          wget https://github.com/jarrodmillman/scipy-ci/raw/windows-openblas/openblas_32_if.zip
```
Comment (tylerjereddy) [inline .github/workflows/windows.yml:53]: Perhaps best deferred to another discussion/issue, but if our OpenBLAS handling is getting a bit more tricky, does this mean we aren't going to be able to depend on `tools/openblas_support.py`.

Two reasons I ask:
- we tend to pull over changes/improvements in that file from NumPy from time to time
- we use it for our wheels infrastructure, and I currently have it involved in the draft `cibuildwheel` infrastructure

Comment (matthew-brett) [reply .github/workflows/windows.yml:53]: Just a flyby comment - but I was planning to work with everyone to get our OpenBLAS / Windows builds updated, and had imagined keeping something like the existing system ...  The binary here was just a get-it-working-fast build I did when porting the Scipy build to Meson / mingw-w64.

**Comment by tylerjereddy** [comment]
Ok, this is basically exactly what was asked for by Ralf in the matching issue, to make CI more reliable, and it makes sense to me too, so I'll merge it.

I believe I'm just slightly out of the loop on why we need that special binary, but sounds like this will be cleared up in the future based on the comments I'm seeing.

**Comment by tylerjereddy** [comment]
thanks

**Comment by mattip** [comment]
Could the repo https://github.com/jarrodmillman/scipy-ci get a README that describes what is there and how it is built?

**Pull request merged**